lib/_io.scm

```diff
@@ -11628,68 +11630,73 @@
         (##table-set! foreign-write-handler-table tag proc)
         (##table-set! foreign-write-handler-table tag))))
 
-(define-prim (##explode-object obj)
-  (##vector-copy obj))
-
-(define-prim (##implode-object re fields subtype)
-  (let* ((n (##vector-length fields))
-         (v (##make-vector n)))
-    (##subtype-set! v subtype)
-    (let loop ((i (##fx- n 1)))
-      (if (##fx< i 0)
-          v
-          (let ((obj (##vector-ref fields i)))
-            (if (##label-marker? obj)
-                (##label-marker-fixup-handler-add!
-                 re
-                 obj
-                 (lambda (resolved-obj)
-                   (##vector-set! v i resolved-obj)))
-                (##vector-set! v i obj))
-            (loop (##fx- i 1)))))))
-
-(define-prim (##explode-structure obj)
-  (##explode-object obj))
+(define-prim (##explode-structure struct)
+  (##declare (not interrupts-enabled))
```

Comment:
In `##explode-structure` and `##implode-structure`, is it OK to disable interrupts,  with the memory allocations?